### PR TITLE
[MIG] [11.0] Migrate hr_public-holidays to 11.0

### PR DESCRIPTION
**Pull request by feketemihai**

### CODE REVIEW

File: hr_public_holidays/README.rst
Comment (elicoidal) [inline hr_public_holidays/README.rst]: remove if empty

File: hr_public_holidays/README.rst
Comment (elicoidal) [inline hr_public_holidays/README.rst]: better use the png file

File: hr_public_holidays/__manifest__.py
Comment (elicoidal) [inline hr_public_holidays/__manifest__.py]: We usually encode the address of the repository here now

**Review by elicoidal — CHANGES_REQUESTED** [review]
some details. Other than that, LGTM

**Comment by pedrobaeza** [comment]
Please merge together "OCA Transbot..." commits https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

**Comment by pedrobaeza** [comment]
I'm now analyzing this module in v10 and how the new feature in vanilla Odoo v11 works, and you should change the approach of the module: you have to overwrite some `resource` methods for taking into account public holidays. That way, you won't need `hr_holidays_compute_days` module, and the rest of the features that use standard resource allocation will be informed about that "public" leaves.

**Comment by feketemihai** [comment]
@pedrobaeza I will adapt it in next days...

**Comment by pedrobaeza** [comment]
Module should be renamed also to `hr_holidays_public` to conform the nomenclature guide and match with the rest of the addons of the same topic.

**Comment by feketemihai** [comment]
@pedrobaeza Can i have you opinion on this, i will rename the module at the end and squash commits.

File: hr_public_holidays/models/hr_public_holidays.py
Comment (astirpe) [inline hr_public_holidays/models/hr_public_holidays.py]: Instead of creating `resource.calendar.leaves` directly in the create() or the write() methods, I would prefer a solution similar to the one we adopted [here](https://github.com/onesteinbv/addons-onestein/blob/10.0/hr_public_holidays_leaves/models/hr_public_holidays.py).

The idea is to create a [Status field](https://github.com/onesteinbv/addons-onestein/blob/10.0/hr_public_holidays_leaves/models/hr_public_holidays.py#L15) in the `hr.holidays.public` model and adopt a minimalistic workflow [`draft`->`validate`](https://github.com/onesteinbv/addons-onestein/blob/10.0/hr_public_holidays_leaves/views/hr_public_holiday.xml#L10-L13).

File: hr_public_holidays/models/hr_public_holidays.py
Comment (astirpe) [inline hr_public_holidays/models/hr_public_holidays.py]: I don't like the idea to use a create() inside an onchange method. You may want to adopt the proposal to use a workflow, this way you don't need this onchange method at all.

**Review by astirpe — CHANGES_REQUESTED** [review]

File: hr_public_holidays/models/hr_public_holidays.py
Comment (astirpe) [inline hr_public_holidays/models/hr_public_holidays.py]: I don't think that unlink() and create() methods called inside an onchange are a good idea. If you adopt the workflow I proposed (see https://github.com/OCA/hr/pull/401/files#r160138849), you will not need this onchange method at all.

**Review by astirpe — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
Any way, I don't think we should create resource leaves, but inherit computation methods for taking into account public holidays, as criteria like states (or you can add cities in this version) can't be reflected in a general resource leave.

**Comment by astirpe** [comment]
The proposal of @pedrobaeza is preferrable.

**Comment by pedrobaeza** [comment]
Check my PR for v10, that takes similar names and structure of the methods of v11 for computing working days and so on.

**Comment by feketemihai** [comment]
@pedrobaeza I thought it will show in calendar as a global holiday, that's why i created a calendar leave...i will change...

**Comment by feketemihai** [comment]
@pedrobaeza Do you think it's worth it to add the leave status parameters (https://github.com/Tecnativa/hr/blob/e9a42112b88bf140921c2779ecc34dcd0ae0ccf2/hr_holidays_compute_days/models/hr_holidays_status.py) since this is the default behaviour on 11 for rest days, and we should do the same for public holidays, only to have the full days configuration..

**Comment by pedrobaeza** [comment]
Well, that should be an extra of course for letting the user to select by type if they want to exclude that, because there are all type of use cases:

* Leaves counted by natural days, not working days. You should check how to mitigate this, as Odoo works now only with working days.
* Jobs that attends emergencies, and thus, there are no public holidays.
* Other combinations...
*

File: hr_public_holidays/models/hr_public_holidays.py
Comment (hieulucky111) [inline hr_public_holidays/models/hr_public_holidays.py]: `holiday = super(HrPublicHolidays, self).write(vals)` return `True`, not object. The loop will raise the error.

File: hr_public_holidays/models/resource.py
Comment (hieulucky111) [inline hr_public_holidays/models/resource.py]: If `leave.tz` is False, `to_tz` function will raise the error. Because user can create public holidays without timezone in `resource.calendar.leaves` record. I think we need to add a control for this.

**Review by hieulucky111 — CHANGES_REQUESTED** [review]

File: hr_public_holidays/models/hr_public_holidays.py
Comment (feketemihai) [reply hr_public_holidays/models/hr_public_holidays.py]: @hieulucky111 Code has been updated since your review comments and the requested change action, please review again

File: hr_public_holidays/models/resource.py
Comment (feketemihai) [reply hr_public_holidays/models/resource.py]: @hieulucky111 Code has been updated since your review comments and the requested change action, please review again

**Comment by feketemihai** [comment]
@pedrobaeza I renamed the module, but is there a way to handle translation files, there you have the link to the module, which remained hr_public_holidays...

**Comment by nguyennghia2909** [comment]
When I create leave by user having only `Employee > Employee` group, I receive error:
 > Sorry, you are not allowed to modify this document. Only users with the following access level are currently allowed to do that: 
> Leaves/Manager 
(Document model: hr.holidays.status)

I think field `show_full_days` should be readonly.

**Comment by feketemihai** [comment]
@nguyennghia2909 Thanks for the hint, updated the view.

**Comment by feketemihai** [comment]
@elicoidal @hieulucky111 @astirpe @pedrobaeza  Can i have your final review on this?

**Comment by feketemihai** [comment]
@pedrobaeza @astirpe @nguyennghia2909 @elicoidal @hieulucky111 Can i have your last review, please.

File: hr_holidays_public/README.rst
Comment (elicoidal) [inline hr_holidays_public/README.rst]: enumerated list is advised here `#. `

File: hr_holidays_public/__manifest__.py
Comment (elicoidal) [inline hr_holidays_public/__manifest__.py]: use the github repo URL

File: hr_holidays_public/models/hr_employee.py
Comment (elicoidal) [inline hr_holidays_public/models/hr_employee.py:15]: should not this be `_compute_`...?

File: hr_holidays_public/models/hr_employee.py
Comment (elicoidal) [inline hr_holidays_public/models/hr_employee.py]: do you need this function to be public?
`compute_`...?

File: hr_holidays_public/models/hr_public_holidays_line.py
Comment (elicoidal) [inline hr_holidays_public/models/hr_public_holidays_line.py]: unclear the purpose of the field

**Review by elicoidal — COMMENTED** [review]
some details

File: hr_holidays_public/models/hr_employee.py
Comment (feketemihai) [reply hr_holidays_public/models/hr_employee.py:15]: _compute_ is used for computed fields, and from there the warning that each compute function of a field should start with "_compute"

File: hr_holidays_public/models/__init__.py
Comment (hieulucky111) [inline hr_holidays_public/models/__init__.py]: I think we should order by alphabet.

File: hr_holidays_public/models/hr_employee.py
Comment (hieulucky111) [inline hr_holidays_public/models/hr_employee.py:6]: Import structure follow OCA guidelines https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#importerror

File: hr_holidays_public/models/hr_holidays.py
Comment (hieulucky111) [inline hr_holidays_public/models/hr_holidays.py]: Import structure follow OCA guidelines https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#importerror

File: hr_holidays_public/models/hr_holidays.py
Comment (hieulucky111) [inline hr_holidays_public/models/hr_holidays.py:75]: `tz_name` still can be none. Can it make the error?

File: hr_holidays_public/report/holidays_summary_report.py
Comment (hieulucky111) [inline hr_holidays_public/report/holidays_summary_report.py:5]: Import structure follow OCA guidelines https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#importerror

File: hr_holidays_public/models/hr_public_holidays_line.py
Comment (feketemihai) [reply hr_holidays_public/models/hr_public_holidays_line.py]: @elicoidal I don't remember why, the field is seems is not used, i think the purpose was that it should allow changing the date when you copy the holidays into a new year...

File: hr_holidays_public/tests/__init__.py
Comment (hieulucky111) [inline hr_holidays_public/tests/__init__.py]: Import order?

File: hr_holidays_public/models/hr_holidays.py
Comment (hieulucky111) [inline hr_holidays_public/models/hr_holidays.py:53]: `tz_name` still can be none. Can it make the error?

File: hr_holidays_public/models/hr_holidays.py
Comment (feketemihai) [reply hr_holidays_public/models/hr_holidays.py]: @hieulucky111 dateutil is not an external dependency, it's included in odoo requirements..

File: hr_holidays_public/models/hr_employee.py
Comment (feketemihai) [reply hr_holidays_public/models/hr_employee.py:6]: same as above

File: hr_holidays_public/models/hr_holidays.py
Comment (feketemihai) [reply hr_holidays_public/models/hr_holidays.py:75]: If not set, dateutil gets the tz from os...

File: hr_holidays_public/models/hr_public_holidays_line.py
Comment (elicoidal) [reply hr_holidays_public/models/hr_public_holidays_line.py]: So just informational?
If possible try to add help or improve the string (nice to have)

**Comment by feketemihai** [comment]
I will squash at the end.

**Comment by feketemihai** [comment]
Guys, can i have a last review here, please

File: hr_holidays_public/README.rst
Comment (elicoidal) [inline hr_holidays_public/README.rst:12]: remove starting period

File: hr_holidays_public/README.rst
Comment (elicoidal) [inline hr_holidays_public/README.rst:19]: better `#. `

File: hr_holidays_public/README.rst
Comment (elicoidal) [inline hr_holidays_public/README.rst:44]: remove if empty

File: hr_holidays_public/__manifest__.py
Comment (elicoidal) [inline hr_holidays_public/__manifest__.py:9]: remove email (already in the README)

File: hr_holidays_public/models/hr_holidays_status.py
Comment (elicoidal) [inline hr_holidays_public/models/hr_holidays_status.py:26]: "If enabled, the number of computed days will always be for the "

File: hr_holidays_public/tests/test_holidays_calculation.py
Comment (elicoidal) [inline hr_holidays_public/tests/test_holidays_calculation.py:151]: remove dead code

File: hr_holidays_public/views/hr_holidays_views.xml
Comment (elicoidal) [inline hr_holidays_public/views/hr_holidays_views.xml:13]: Improve the format for readability

File: hr_holidays_public/views/hr_holidays_views.xml
Comment (elicoidal) [inline hr_holidays_public/views/hr_holidays_views.xml:74]: Same here

**Review by elicoidal — CHANGES_REQUESTED** [review]
formating details on my side

**Comment by pedrobaeza** [comment]
I want to review this to sync with the PRs I have in v10, but this week is overloaded for me. I will try to do it ASAP.

**Review by MiquelRForgeFlow — COMMENTED** [review]
I think all the files named <code>public_holidays_X</code> should be renamed <code>holidays_public_X</code> and also all the classes named <code>PublicHolidaysX</code> should be renamed <code>HolidaysPublicX</code>.

**Comment by MiquelRForgeFlow** [comment]
@feketemihai how are you? could you attend last comments?

File: hr_holidays_public/models/hr_holidays.py
Comment (novawish) [inline hr_holidays_public/models/hr_holidays.py]: The original function's name had been changed (here)[https://github.com/odoo/odoo/commit/6de803645e9802019f122ac7e5d9abd18a71e742]. Can you update it?

Comment (novawish) [inline hr_holidays_public/models/hr_holidays.py:45]: The original function's name had been changed [here](https://github.com/odoo/odoo/commit/6de803645e9802019f122ac7e5d9abd18a71e742). Can you update it?

**Review by novawish — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
Superseeded by #465

**Pull request closed without merging**
